src/Server/HTTP/authenticateUserByHTTP.cpp

```diff
@@ -0,0 +1,269 @@
+#include <Server/HTTP/authenticateUserByHTTP.h>
+
+#include <Access/Authentication.h>
+#include <Access/Common/SSLCertificateSubjects.h>
+#include <Access/Credentials.h>
+#include <Access/ExternalAuthenticators.h>
+#include <Server/HTTP/HTTPServerRequest.h>
+#include <Server/HTTP/HTMLForm.h>
+#include <Server/HTTP/HTTPServerResponse.h>
+#include <Interpreters/Context.h>
+#include <Interpreters/Session.h>
+
+#include <Poco/MemoryStream.h>
+#include <Poco/Base64Decoder.h>
+#include <Poco/Base64Encoder.h>
+#include <Poco/StreamCopier.h>
+#include <Poco/Net/HTTPBasicCredentials.h>
+
+#if USE_SSL
+#include <Poco/Net/X509Certificate.h>
+#endif
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int AUTHENTICATION_FAILED;
+    extern const int BAD_ARGUMENTS;
+    extern const int SUPPORT_IS_DISABLED;
+}
+
+
+namespace
+{
+    String base64Decode(const String & encoded)
```

Comment:
`Common/Base64.h`

Reply:
Corrected, now `Common/Base64.h` is used